[docs][explore] Setup for demo app (#15072)
yugabyte/yugabyte-db: docs/content/preview/explore/_index.md

  </div>
</div>

## Set up YB Simulation Base Demo

YB Simulation Base Demo is a Java application that simulates workloads against YugabyteDB. Some Explore topics use the application to demonstrate features of YugabyteDB.

> Download the Simulation Base Demo JAR file (`yb-simu-base-app.jar`) using the following command:

Reviewer: ```suggestion
Download the Simulation Base Demo JAR file (`yb-simu-base-app-0.0.1.jar`) using the following command:
```

```sh

> wget https://github.com/yugabyte/yb-simulation-base-demo-app/releases/download/1.0/yb-simu-base-app.jar

Reviewer: ```suggestion
wget https://github.com/yugabyte/yb-simulation-base-demo-app/releases/download/v0.0.1/yb-simu-base-app-0.0.1.jar
```

```

To start the application against a running local cluster, use the following command:

```sh

> java -jar ./yb-simu-base-app.jar

Reviewer: ```suggestion
java -jar ./yb-simu-base-app-0.0.1.jar
```

Reviewer: Note that you need Java 11 or later to run the application.

```

> By default, the application connects to the local cluster at 127.0.0.1, adds tables, seeds some data, and starts a workload.

Reviewer: ```suggestion
By default, the application connects to the local cluster at 127.0.0.1 and from the UI, you can add tables, seed some data, and start a workload.
```

Reviewer: This statement gives me a sense that everything is taken care of when we start the jar.

To connect to a different address or node, use the -Dnode flag to specify an IP address. For example:

```sh
java -jar ./yb-simu-base-app.jar -Dnode=127.0.0.2
```

> To view the application UI, navigate to <http://localhost:8080>. to view the generic workload running on the local cluster.

Reviewer: ```suggestion
To view the application UI, navigate to <http://localhost:8080>. You can view the network diagram with the cluster details, latency and throughput graphs for the generic workloads running on the local cluster.
Note that the graphs are empty when you load the page and starts populating when you run a workload.
```

### Install

YB Simulation Base Demo requires Java 11 or later installed on your computer.

Download the Simulation Base Demo JAR file (`yb-simu-base-app.jar`) using the following command:

```sh
wget https://github.com/yugabyte/yb-simulation-base-demo-app/releases/download/1.0/yb-simu-base-app.jar
```

### Run

To start the application against a running local cluster, use the following command:

```sh
java -jar ./yb-simu-base-app.jar
```

By default, the application connects to the local cluster at 127.0.0.1.

To connect to a different address or node, use the `-Dnode` flag to specify an IP address. For example:

```sh
java -jar ./yb-simu-base-app.jar -Dnode=127.0.0.2
```

To view the application UI, navigate to <http://localhost:8080>.

### Start a workload

To start a workload, do the following:

1. In the application UI, Click the hamburger icon beside Active Workloads for Generic.
1. Select the **Usable Operations** tab.
1. Under **Create Tables**, click **Run Create Tables Workload** to add tables to the database.
1. Under **Seed Data**, click **Run Seed Data Workload** to add data to the tables.

> 1. Under **Simulation**, click **Run Simulation Workload**.

Reviewer: We need to enable **Include new Inserts** and then **Run Simulation Workloads**. By default the app performs only Reads.

Reviewer: Under **Simulation**, enable **Include new Inserts** and select **Run Simulation Workload** to start a simulation that performs read and write operations across all the nodes of your cluster.

  </div>
</div>

## Set up YB Workload Simulator

YB Workload Simulator is a Java application that simulates workloads against YugabyteDB and provides live metrics of latency and throughput from the application's point of view. Some Explore topics use the application to demonstrate features of YugabyteDB. 

The application uses the YugabyteDB JDBC [Smart Driver](../drivers-orms/smart-drivers/), which features cluster- and topology-aware connection load balancing. For more information about the workload application, refer to [YB Workload Simulator](https://github.com/YugabyteDB-Samples/yb-workload-simulator/).

### Download

YB Workload Simulator requires Java 11 or later installed on your computer. {{% jdk-setup %}}

> Download the YB Workload Simulator JAR file (`yb-simu-base-app.jar`) using the following command:

Reviewer: ```suggestion
Download the YB Workload Simulator JAR file using the following command:
```

Reviewer: I don't think we need to specify the name.

```sh

> wget https://github.com/YugabyteDB-Samples/yb-workload-simulator/releases/download/1.0/yb-simu-base-app.jar

Reviewer: ```suggestion
wget https://github.com/YugabyteDB-Samples/yb-workload-simulator/releases/download/v0.0.2/yb-workload-sim-0.0.2.jar
```

       role="tab" aria-controls="cloud" aria-selected="true">
      <i class="fas fa-cloud" aria-hidden="true"></i>
      Use a cloud cluster
    </a>
  </li>
  <li>
    <a href="#localworkload" class="nav-link" id="local-tab" data-toggle="tab"
       role="tab" aria-controls="local" aria-selected="false">
      <i class="icon-shell" aria-hidden="true"></i>
      Use a local cluster
    </a>
  </li>
</ul>

<div class="tab-content">
  <div id="cloudworkload" class="tab-pane fade show active" role="tabpanel" aria-labelledby="cloud-tab">

To connect the application to your cluster, ensure that you have downloaded the cluster SSL certificate and your computer is added to the IP allow list. Refer to [Before you begin](../develop/build-apps/cloud-add-ip/).

To start the application against a running YugabyteDB Managed cluster, use the following command:

```sh
java -Dnode=<host name> \
    -Ddbname=<dbname> \
    -Ddbuser=<dbuser> \
    -Ddbpassword=<dbpassword> \
    -Dspring.datasource.hikari.data-source-properties.topologyKeys=<cloud.region.zone>
    -Dssl=true \
    -Dsslmode=verify-full \
    -Dsslrootcert=<path-to-cluster-certificate> \

>     -jar ./yb-simu-base-app.jar

Reviewer: ```suggestion
    -jar ./yb-workload-sim-0.0.2.jar
```

- `<cloud.region.zone>` - the zones in your cluster, comma-separated, in the format `cloud.region.zone`, to be used as topology keys for [topology-aware load balancing](../drivers-orms/smart-drivers/#topology-aware-connection-load-balancing). For example, to add topology keys for a multi-zone cluster in the AWS US East region, you would enter the following:

    ```sh
    -Dspring.datasource.hikari.data-source-properties.topologyKeys=aws.us-east.us-east-1a,aws.us-east.us-east-2a,aws.us-east.us-east-3a
    ```

- `<path-to-cluster-certificate>` with the path to the [cluster certificate](../yugabyte-cloud/cloud-secure-clusters/cloud-authentication/) on your computer.

  </div>
  <div id="localworkload" class="tab-pane fade" role="tabpanel" aria-labelledby="local-tab">

To start the application against a running local cluster, use the following command:

```sh
java -jar \
    -Dnode=127.0.0.1 \
    -Dspring.datasource.hikari.data-source-properties.topologyKeys=aws.us-east.us-east-1a,aws.us-east.us-east-2a,aws.us-east.us-east-3a \

>     ./yb-simu-base-app.jar

Reviewer: ```suggestion
    ./yb-workload-sim-0.0.2.jar
```

  </div>
</div>

> ## Set up YB Workload Simulator

Reviewer: I thought we decided on YugabyteDB Workload Simulator.
